### PR TITLE
Bugfix/blog multi authors

### LINKED ISSUE #5328: BlogPost should account for multiple authors
Some blogposts have multiple authors. Now that author urls are utilized, this creates an odd UX where both authors are highlighted in a link to one source.

Adjust the BlogPost cards to account for multiple authors, creating separate links for each one.

**Error example:**
`/blog/real-weakest-link-in-software-supply-chain-security/`

Image

`/blog/remote-software-development-made-easier/`

Image

### PR DESCRIPTION
**Pull request by st0nebraker**
Closes #5328 to allow multiple authors on blogposts along with the authors url.

### NOTE
- Focus on these 4 files. The other 400+ are just reference updates
Screen Shot 2022-05-17 at 1 04 38 PM

- Accompanying PR to update Handbook blog post instructions: Handbook [PR#3491](https://github.com/sourcegraph/handbook/pull/3491)
- Happy to sit on this PR for a few days if the team thinks it'll be too cumbersome to deal with the merge!

### TEST
- Nav to `/blog/real-weakest-link-in-software-supply-chain-security/` and ensure multiple blog authors are supported
- Ensure no breaking changes for other blog posts

**BEFORE:**
image

**NOW:**
Screen Shot 2022-05-17 at 1 17 36 PM

### CODE REVIEW

File: website/src/components/blog/BlogPost.tsx [hunk 1]
```diff
@@ -46,23 +46,39 @@ export const BlogPost: React.FunctionComponent<Props> = ({
...
+                {post.frontmatter.authors?.length && (
                     <p className="blog-post__byline mb-0">
-                        {post.frontmatter.authorUrl ? (
-                            post.frontmatter.authorUrl.includes('http') ? (
-                                <a href={post.frontmatter.authorUrl} target="_blank" rel="nofollow noreferrer">
-                                    {post.frontmatter.author}
-                                </a>
-                            ) : (
-                                <Link to={post.frontmatter.authorUrl}>{post.frontmatter.author}</Link>
+                        {post.frontmatter.authors.map((a, index) => {
```
Comment (zlonko) [inline website/src/components/blog/BlogPost.tsx:51]: I tested several byline cases and found that your solution is super robust! No unexpected styles here:

One author with link:
Screen Shot 2022-05-17 at 3 33 49 PM

Multiple authors with links:
Screen Shot 2022-05-17 at 3 33 44 PM

One author without a link:
Screen Shot 2022-05-17 at 4 02 11 PM

Multiple authors without links:
Screen Shot 2022-05-17 at 3 33 37 PM

**Review by zlonko — DISMISSED** [review]
This looks awesome. I left a few notes, but they aren't mission critical. Let me know what you think!

File: website/src/components/blog/BlogPost.tsx [hunk 2]
```diff
@@ -46,23 +46,39 @@ export const BlogPost: React.FunctionComponent<Props> = ({
                         post.frontmatter.title
                     )}
                 </h1>
-                {post.frontmatter.author && post.frontmatter.publishDate && (
+                {post.frontmatter.authors?.length && (
                     <p className="blog-post__byline mb-0">
```
[the remark this answers is not in the document]
Comment (st0nebraker) [reply website/src/components/blog/BlogPost.tsx]: Good call!

[the remark this answers is not in the document]
Comment (st0nebraker) [reply website/src/components/blog/BlogPost.tsx]: Good call!

File: website/src/components/blog/BlogPost.tsx [hunk 3]
```diff
@@ -46,23 +46,39 @@ export const BlogPost: React.FunctionComponent<Props> = ({
...
+                                        ) : (
+                                            <Link to={a.url}>{a.name}</Link>
+                                        )
+                                    ) : (
+                                        a.name
+                                    )}
+                                    {post.frontmatter.authors!.length > 1 &&
+                                    index !== post.frontmatter.authors!.length - 1
+                                        ? ', '
+                                        : ' '}
```
[the remark this answers is not in the document]
Comment (st0nebraker) [reply website/src/components/blog/BlogPost.tsx]: Smart! Put it in. One less variable to reference 🤓

File: website/src/components/blog/BlogPost.tsx [hunk 4]
```diff
@@ -46,23 +46,39 @@ export const BlogPost: React.FunctionComponent<Props> = ({
                         post.frontmatter.title
                     )}
                 </h1>
-                {post.frontmatter.author && post.frontmatter.publishDate && (
+                {post.frontmatter.authors?.length && (
```
[the remark this answers is not in the document]
Comment (st0nebraker) [reply website/src/components/blog/BlogPost.tsx:49]: Thanks!

File: website/src/components/blog/BlogPost.tsx (hunk 1 above)
Comment (st0nebraker) [reply website/src/components/blog/BlogPost.tsx:51]: Woohoo 😸

**Pull request closed without merging**